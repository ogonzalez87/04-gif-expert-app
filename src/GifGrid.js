import React, { Fragment, useEffect, useState } from "react";
import { GifGridItem } from "./components/GifGridItem";
import { getGifs } from "./helpers/getGifs";

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs(category)
    .then( setImages ) ;
  }, [category]);

  

  return (
    
    <Fragment>
      <h3> {category} </h3>
      <div className="card-grid">
  
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </Fragment>
  );
};
