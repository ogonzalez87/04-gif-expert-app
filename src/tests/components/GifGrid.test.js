import React from "react";
import { shallow } from "enzyme";
import { GifGrid } from "../../GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs")


describe("GifGrid test", () => {
  const category = "goku";
  test("should load component", () => {

    const gifs = [{ 
        id: 'adc',
        url: 'google.com',
        title : 'cosa'
    }]

    useFetchGifs.mockReturnValue({})
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should  show items on useFetchGifs", () => {

     
    const wrapper = shallow(<GifGrid category={category} />); 
  });
});
