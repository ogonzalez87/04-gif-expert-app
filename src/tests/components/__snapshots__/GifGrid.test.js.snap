// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid test should  show items on useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    goku
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid test should load component 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    goku
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
